packages/p2p/src/components/order-details/__tests__/order-details-wrapper.spec.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { screen, render, waitFor } from '@testing-library/react';
+import { useStores } from 'Stores';
+import { isMobile } from '@deriv/shared';
+import OrderDetailsWrapper from '../order-details-wrapper';
+
+const MockOrderWrapper = props => {
+    return isMobile() ? (
+        <OrderDetailsWrapper {...props}>
+            <p>desktop test</p>
+        </OrderDetailsWrapper>
+    ) : (
+        <OrderDetailsWrapper {...props}>
+            <p>mobile test</p>
+        </OrderDetailsWrapper>
+    );
+};
+
+jest.mock('Stores', () => ({
+    ...jest.requireActual('Stores'),
+    useStores: jest.fn(),
+}));
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isMobile: jest.fn(),
+}));
+
+const mockUseStores = () => {
+    useStores.mockImplementation(() => ({
+        order_store: { order_information: { should_show_order_footer: true } },
+        sendbird_store: { setShouldShowChatModal: false },
+    }));
+};
+
+beforeAll(() => mockUseStores());
+
+describe('<OrderDetailsWrapper />', () => {
+    it('Should render component on desktop version', async () => {
+        isMobile.mockReturnValue(true);
+        const desktop_props = {
+            onClick: jest.fn(),
+            page_title: 'some text',
+        };
+
+        render(<MockOrderWrapper {...desktop_props} />);
+
+        await waitFor(() => {
+            expect(screen.getByTestId('order-details-wrapper-mobile')).toBeInTheDocument();
+            expect(screen.getByText('desktop test')).toBeInTheDocument();
+        });
+    });
+    it('Should render component on mobile version', async () => {
+        isMobile.mockReturnValue(false);
+        const mobile_props = {
+            className: 'order-details',
+            body_className: 'order-details--body',
+            height_offset: '80px',
+            is_flex: true,
+            is_modal_open: true,
+        };
+
+        render(<MockOrderWrapper {...mobile_props} />);
+
+        await waitFor(() => {
+            expect(screen.getByTestId('order-details-wrapper-desktop')).toBeInTheDocument();
+            expect(screen.getByText('mobile test')).toBeInTheDocument();
```

Comment:
What about checking the chat icon for mobile view?

Reply:
For now resolved as `expect(container.getElementsByClassName('dc-icon').length).toBe(1)`
And `order_store: { order_information: { should_show_order_footer: true } }`

Have no idea how else we can test footer. Please advice something :pray:

Reply:
"order_store: { order_information: { should_show_order_footer: true } }" With this line a footer will get rendered inside your mobile wrapper element (<OrderDetailsFooter order_information={order_store.order_information} />).
 But we are not checking that element is present or not in the test case, I think we can push dummy values in to "order_information={order_store.order_information} " these object and check the presence of different elements in the footer component [(link)](https://github.com/binary-com/deriv-app/blob/8f536b15c2a4a5b81300da92d502b461ba57dc64/packages/p2p/src/components/order-details/order-details-footer.jsx) accordingly.

Reply:
If you have any doubt on this , you can reach me out. I think I can help.